engine/flink/util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/aggregate/AggregateTransformer.scala

```diff
@@ -0,0 +1,127 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.aggregate
+
+import org.apache.flink.api.common.functions.RichMapFunction
+import org.apache.flink.api.common.state.ValueStateDescriptor
+import org.apache.flink.streaming.api.functions.KeyedProcessFunction
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.streaming.api.windowing.assigners.TumblingProcessingTimeWindows
+import org.apache.flink.streaming.api.windowing.time.Time
+import org.apache.flink.util.Collector
+import pl.touk.nussknacker.engine.api.context.ContextTransformation
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.NodeId
+import pl.touk.nussknacker.engine.api.{Context => NkContext, _}
+import pl.touk.nussknacker.engine.flink.api.process._
+import pl.touk.nussknacker.engine.flink.api.state.LatelyEvictableStateFunction
+
+import scala.collection.immutable.TreeMap
+import scala.concurrent.duration.Duration
+
+object AggregateTransformer extends CustomStreamTransformer {
+
+  @MethodToInvoke(returnType = classOf[AnyRef])
+  def execute(@ParamName("keyBy") keyBy: LazyParameter[String],
+              @ParamName("length") length: String,
+              @ParamName("aggregator") aggregator: Aggregator,
```

Comment:
How do you want to pass `aggregator`? Some map of all possible values in global variables? WDYT about replacing it with java Enum and make it more standardized and ready for further better enum handling?

Reply:
I refactored it a bit, to be able to separate
- aggregators
- functions/transformers 
- UI/API - that is, how do we present it to user, what is fixed and what can be changed